packages/api/src/command/hie/unlink-patient-from-organization.ts

```diff
@@ -0,0 +1,379 @@
+    const preConversionFileName = buildDocumentNameForPreConversion(fileName);
+
+    const fileNames = [
+      conversionResultFileName,
+      normalizedFileName,
+      cleanFileName,
+      fromConverterFileName,
+      preConversionFileName,
+    ];
+
+    const partialConversionFileNames = await getPartialConversionFileNames(preConversionFileName);
+    const allFileNames = [...fileNames, ...partialConversionFileNames];
+
+    const existingFiles = await Promise.all(
+      allFileNames.map(async fileName => {
+        try {
+          const exists = await s3Utils.fileExists(s3ConversionResultBucketName, fileName);
+          return exists ? fileName : undefined;
+        } catch (err) {
+          return undefined;
+        }
+      })
+    );
+
+    const validFiles = existingFiles.filter(fileName => fileName !== undefined) as string[];
+
+    for (const fileName of validFiles) {
+      log(`${dryRunMsg}Deleting file ${fileName} from S3`);
+      if (!dryRun) {
+        await s3Utils.deleteFile({ bucket: s3ConversionResultBucketName, key: fileName });
+      }
+    }
+  } catch (error) {
+    log("Error removing conversion results from S3:", errorToString(error));
+    throw error;
+  }
+}
+
+async function getPartialConversionFileNames(preConversionFileName: string): Promise<string[]> {
+  const partialConversionFileNames: string[] = [];
+  let index = 0;
+  let partialExists = true;
+  while (partialExists) {
+    const partialFileName = buildDocumentNameForPartialConversions(preConversionFileName, index);
+    try {
+      partialExists = await s3Utils.fileExists(s3ConversionResultBucketName, partialFileName);
+      if (partialExists) {
+        partialConversionFileNames.push(partialFileName);
+      }
+    } catch (err) {
+      partialExists = false;
+    }
+    index++;
+  }
+  return partialConversionFileNames;
+}
+
+async function findAndRemoveMedicalDocumentFromS3(
+  fileName: string,
+  dryRun = false,
+  log: typeof console.log
+): Promise<void> {
+  const dryRunMsg = getDryRunPrefix(dryRun);
+  try {
+    const documentExists = await s3Utils.fileExists(s3MedicalDocumentsBucketName, fileName);
+
+    if (!documentExists) return;
+
+    log(`${dryRunMsg}Deleting file ${fileName} from S3`);
+    if (!dryRun) {
+      await s3Utils.deleteFile({ bucket: s3MedicalDocumentsBucketName, key: fileName });
+    }
+  } catch (error) {
+    log("Error removing medical document from S3:", errorToString(error));
+    throw error;
+  }
+}
+
+async function findAndRemoveConsolidatedDocumentFromS3(
+  cxId: string,
+  patientId: string,
+  dryRun = false,
+  log: typeof console.log
+): Promise<void> {
+  const dryRunMsg = getDryRunPrefix(dryRun);
+  try {
+    const consolidatedPrefix = `${createFolderName(
+      cxId,
+      patientId
+    )}/${cxId}_${patientId}_consolidated`;
+    const medicalRecordsPrefix = `${createFolderName(cxId, patientId)}/${cxId}_${patientId}_MR`;
```

Comment:
Let's also move these to functions, please? I love what you did with the `buildDocumentNameFor...()` functions! 😄